tasks/go_test.py

```diff
@@ -733,3 +737,145 @@ def parse_test_log(log_file):
+                    continue
+                impacted_modules.add(module)
+                stack.extend(dependencies[module])
+            cache[modified_module] = impacted_modules
+    return impacted_modules
+
+
+def format_packages(ctx, impacted_packages):
+    """
+    Format the packages list to be used in our test function. Will take each path and create a list of modules with its targets
+    """
+    # if build_tags is None:
+    build_tags = []
+
+    packages = [f'{package.replace("github.com/DataDog/datadog-agent/", "./")}' for package in impacted_packages]
+    modules_to_test = {}
+    go_mod_modified_modules = set()
+    for package in packages:
+        match_precision = 0
+        best_module_path = None
+
+        # Since several modules can match the path we take only the most precise one
+        for module_path in DEFAULT_MODULES:
+            package_path = Path(package)
+            if package_path.is_relative_to(module_path):
+                if len(module_path) > match_precision:
+                    match_precision = len(module_path)
+                    best_module_path = module_path
+
+        # Check if the package is in the target list of the module we want to test
+        targeted = False
+        for target in DEFAULT_MODULES[best_module_path].targets:
+            if os.path.normpath(os.path.join(best_module_path, target)) in package:
+                targeted = True
+                break
+        if not targeted:
+            continue
+
+        # If go mod was modified in the module we run the test for the whole module so we do not need to add modified packages to targets
+        if best_module_path in go_mod_modified_modules:
+            continue
+
+        # If the package has been deleted we do not try to run tests
+        if not os.path.exists(package):
+            continue
+
+        relative_target = "./" + os.path.relpath(package, best_module_path)
+
+        if best_module_path in modules_to_test:
+            if (
+                modules_to_test[best_module_path].targets is not None
+                and package not in modules_to_test[best_module_path].targets
+            ):
+                modules_to_test[best_module_path].targets.append(relative_target)
+        else:
+            modules_to_test[best_module_path] = GoModule(best_module_path, targets=[relative_target])
+
+    # Clean up duplicated paths to reduce Go test cmd length
+    for module in modules_to_test:
+        modules_to_test[module].targets = clean_nested_paths(modules_to_test[module].targets)
+        if (
+            len(modules_to_test[module].targets) >= WINDOWS_MAX_PACKAGES_NUMBER
+        ):  # With more packages we can reach the limit of the command line length on Windows
+            modules_to_test[module].targets = DEFAULT_MODULES[module].targets
+
+    # Clean up to avoid running tests on package with no Go files matching build tags
+    for module in modules_to_test:
+        print(
```

Comment:
❓ Do we need this print if you launch your invoke task with a `-e` for instance?

Reply:
I will remove it!